security/fma-kona-asterisc.md

```diff
@@ -0,0 +1,213 @@
+
+Stage 1.4 also requires the integration of the Kona + Asterisc Dispute Game into the [op-challenger] offchain
+fault proof component. The Failure Modes and Recovery Paths below detail this as well.
+
+Below are references for this project:
+
+- ["Stage 1.4" internal project document](https://www.notion.so/oplabs/Stage-1-4-Partway-a6f57ad777b148dda01488f2646cff17)
+- [Kona Documentation](https://anton-rs.github.io/kona/).
+- [Kona Repository](https://github.com/anton-rs/kona).
+- [Asterisc Repository](https://github.com/ethereum-optimism/asterisc)
+
+[kona]: https://github.com/anton-rs/kona
+[asterisc]: https://github.com/ethereum-optimism/asterisc
+[cannon]: https://github.com/ethereum-optimism/optimism/tree/develop/cannon
+[op-program]: https://github.com/ethereum-optimism/optimism/tree/develop/op-program
+
+
+## Failure Modes and Recovery Paths
+
+### Kona and Asterisc Hot Path Failure
+
+- **Description:**
+  The `op-program` + `cannon` dispute game breaks and the `kona` + `asterisc` fault dispute game is placed in the hot path.
+  If `kona` + `asterisc` also fail, the permissioned fallback will need to be activated.
+
+- **Risk Assessment:**
+  Low likelihood, high impact.
+  Low likelihood since `op-program` + `cannon` are already running in production and have meaningful improvements on the way that will increase resilience.
+  If `kona` + `asterisc` fails while in the hotpath, the impact is high since the permissioned fallback will need to be triggered.
+  This engages the security council, requires comms, and causes the chain to fallback to a weaker security model.
+
+- **Mitigations:**
+  Offline testing with the VM Runner for both `kona` + `asterisc` as well as `op-program` + `cannon`.
+  The VM Runner also fully tests the integration of the `op-challenger` with these dispute games ([covered in a later section](#Breaking-Backwards-Compatibility-for-the-op-challenger)).
+  Action tests ensuring both proof systems follow spec.
+
+- **Detection:**
+  The game that broke `op-program` + `cannon` would be invalidated since the respected game type would be switched in the dispute game factory.
+  Once `kona` + `asterisc` is the respected game type, the dispute monitor will watch for all new games and will alert if there is an issue with `kona` + `asterisc`.
+  If `kona` + `asterisc` has an issue while in the hot path, and an invalid output / game is played, the dispute monitor and `op-challenger` would alert and detect invalid games.
+  At this point, the permissioned fallback would need to be engaged since both proof systems are in repair.
+
+- **Recovery Path(s):**
+  Likely the only viable recovery path would be the permissioned fallback unless the `op-program` + `cannon` are fixed and able to be placed back into the hotpath.
+
+
+### Divergent Kona Derivation Pipeline
+
+- **Description:**
+  The `kona` derivation pipeline diverged in behavior from the `op-program`'s derivation pipeline.
```

Comment:
Is the implication here that op-program's pipeline is the spec? Or is it possible they diverge because kona is correct and op-program has a bug or doesn't properly implement the spec?

Reply:
Yes, it is the "spec." The derivation specs are mostly incomplete on https://specs.optimism.io, so it is treated as the reference.

`op-program` can diverge from the written spec, but at least at the moment, that would become canonical. This has already happened, actually, with a receipts root bug in `op-geth` in the Regolith hardfork.